src/client/datascience/errors/errorHandler.ts

```diff
@@ -256,16 +289,62 @@ export class DataScienceErrorHandler implements IDataScienceErrorHandler {
             if ((purpose === 'restart' || purpose === 'start') && handler) {
                 await handler(err, defaultErrorMessage);
             } else {
+                void this.displayErrorsInCell(defaultErrorMessage, cellToDisplayErrors);
                 this.applicationShell.showErrorMessage(defaultErrorMessage).then(noop, noop);
             }
         } else {
             // Some errors have localized and/or formatted error messages.
-            this.applicationShell
-                .showErrorMessage(getCombinedErrorMessage(errorPrefix, err.message || err.toString()))
-                .then(noop, noop);
+            const message = getCombinedErrorMessage(errorPrefix, err.message || err.toString());
+            void this.displayErrorsInCell(message, cellToDisplayErrors);
+            this.applicationShell.showErrorMessage(message).then(noop, noop);
         }
         traceError('DataScience Error', err);
     }
+    private async displayErrorsInCell(errorMessage: string, cellToDisplayErrors?: NotebookCell) {
+        if (!cellToDisplayErrors || !errorMessage) {
+            return;
+        }
+        const associatedkernel = this.serviceContainer
+            .get<IKernelProvider>(IKernelProvider)
+            .get(cellToDisplayErrors.notebook);
+        if (!associatedkernel) {
+            return;
+        }
+        // Sometimes the cells are still running, wait for 1s for cells to finish & get cleared,
+        // Then display the error in the cell.
+        const stopWatch = new StopWatch();
+        while (stopWatch.elapsedTime <= 1_000 && associatedkernel.hasPendingCells) {
+            await sleep(100);
+        }
+        if (associatedkernel.hasPendingCells) {
+            return;
+        }
+        const controllers = this.serviceContainer.get<INotebookControllerManager>(INotebookControllerManager);
+        const controller = controllers.getSelectedNotebookController(cellToDisplayErrors.notebook);
+        // Possible it changed.
+        if (!controller || controller.connection !== associatedkernel.kernelConnectionMetadata) {
+            return;
+        }
+        const regex = /\[(?<name>.*)\]\((?<command>command:\S*)\)/gm;
+        let matches: RegExpExecArray | undefined | null;
+        while ((matches = regex.exec(errorMessage)) !== null) {
+            if (matches.length === 3) {
+                errorMessage = errorMessage.replace(matches[0], `<a href='${matches[2]}'>${matches[1]}</a>`);
+            }
+        }
+        const execution = controller.controller.createNotebookCellExecution(cellToDisplayErrors);
```

Comment:
I know this is how the API works, just feels odd that we have to do all this just to change the output. Though I guess it does ensure that everything runs through the execution control path.